rust/chains/hyperlane-cosmos/src/providers/grpc.rs

```diff
@@ -534,6 +540,42 @@ impl WasmProvider for WasmGrpcProvider {
         Ok(response.data)
     }
 
+    async fn wasm_contract_info(&self) -> ChainResult<ContractInfo> {
```

Comment:
if you add a getter for `self.contract_address` on the trait, you could move this implementation inside the trait definition (to have it be the default implementation), since I doubt this method would ever need to be implemented differently

Reply:
I would go in an opposite direction here: hide `get_contract_address` method inside `WasmGrpcProvider` and reuse it in all other methods of `WasmProvider` implemented by `WasmGrpcProvider`. This way the interface of trait `WastProvider` stays clean. I added a commit to illustrate it.

Reply:
Discussed and agreed to remove unnecessary methods from the trait.